src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,100 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = .75f;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        float threshold = table.length * LOAD_FACTOR;
+
+        if (size > threshold) {
+            resize();
+        }
 
+        int hashCode = (key == null ? 0 : key.hashCode());
+        int index = getIndex(hashCode, table.length);
+        Node<K, V> currentNode = table[index];
+
+        if (!replaceValue(currentNode, key, value)) {
+            Node<K, V> newNode = new Node<>(key, value);
+            newNode.next = currentNode;
+            table[index] = newNode;
+            size++;
+        }
```

Comment:
```suggestion
        while (node != null) {
            K currentKey = node.key;
            if (currentKey == key || (currentKey != null && currentKey.equals(key))) {
                node.value = value;
                return;
            }
            node = node.next;
        }
        
        Node<K, V> newNode = new Node<>(key, value);
        newNode.next = currentNode;
        table[index] = newNode;
        size++;
```
 ``` if (!replaceValue(currentNode, key, value)) { ```
this line is a little bit confusing, I would not change anything when checking if the key is present
My suggested code has the same idea, but it is easier to read